src/main/java/com/herocc/bukkit/multispawn/events/PlayerDeath.java

```diff
@@ -4,22 +4,24 @@
 import org.bukkit.entity.Player;
 import org.bukkit.event.EventHandler;
 import org.bukkit.event.Listener;
-import org.bukkit.event.entity.PlayerDeathEvent;
+import org.bukkit.event.player.PlayerRespawnEvent;
 
 public class PlayerDeath implements Listener {
   private final MultiSpawn plugin = MultiSpawn.getPlugin();
 
   @EventHandler
   @SuppressWarnings("unused")
-  public void onPlayerDeath(PlayerDeathEvent ev) {
-    final Player p = ev.getEntity();
-    if (p.hasPermission("multispawn.noteleport") // If player is excluded
+  public void onPlayerDeath(PlayerRespawnEvent ev) {
+    final Player p = ev.getPlayer();
+    if (p.hasPermission("multispawn.noteleportondeath") // If player is excluded
```

Comment:
Could you make this `p.hasPermission("multispawn.noteleportondeath") || p.hasPermission("multispawn.noteleport)"` for backwards compatibility?

Reply:
There will be no option to make players teleport on death but not teleport on join. If you want to teleport on death, you'll need to remove both multispawn.noteleport and multispawn.noteleportondeath, so it will teleport on join, and this may be quite hard to understand. #6 has teleportOnDeath disabled by default, it should help with backwards compatibility.

Reply:
Fair enough. Thinking about it now there would be no breaking changes, considering the feature is already broken haha.